src/main/java/org/example/service/AnimalService.java

```diff
@@ -0,0 +1,77 @@
+package org.example.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.example.model.Animal;
+
+import java.io.*;
+import java.util.*;
+
+public class AnimalService {
```

Comment:
you can create JsonMapper jsonMapper = new JsonMapper(); and same as a class variable, and use it in a different method, instead of creating it all the time when you use the method